tests/topotests/bgp_as_allow_in/test_bgp_as_allow_in.py

```diff
@@ -0,0 +1,1008 @@
+               }
+           }
+        }
+    }
+    result = create_router_bgp(tgen, topo, input_dict_1)
+    assert result is True, "Testcase {} : Failed \n Error: {}".format(
+        tc_name, result)
+    step("We should see R1 advertised prefix only in IPv6 AFI "
+         "not in IPv4 AFI.")
+    result = verify_rib(tgen, 'ipv4', dut, static_route_ipv4,
+                        protocol=protocol, expected=False)
+    assert result is not True, "Testcase {} : Failed \n"
+    "Expected behavior: routes should not be present in ipv4 rib\n"
+    " Error: {}".format(tc_name, result)
+    result = verify_rib(tgen, 'ipv6', dut, static_route_ipv6,
+                        protocol=protocol)
+    assert result is True, "Testcase {} : Failed \n Error: {}".format(
+        tc_name, result)
+
+    write_test_footer(tc_name)
+
+
+def test_bgp_allowas_in_no_of_occurrences_p0(request):
+    """
+    Verify that the if number of occurrences of AS number in path is
+    more than the configured allowas-in value then we do not accept
+    the route.
+
+    """
+
+    tc_name = request.node.name
+    write_test_header(tc_name)
+    tgen = get_topogen()
+    reset_config_on_routers(tgen)
+
+    # Don't run this test if we have any failure.
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    dut = "r3"
+    protocol = "bgp"
+
+    for addr_type in ADDR_TYPES:
+        # Enable static routes
+        static_routes = {
+            "r1": {
+                "static_routes": [
+                    {
+                        "network": NETWORK[addr_type],
+                        "next_hop": NEXT_HOP_IP[addr_type]
+                    }
+                ]
+            }
+        }
+
+        logger.info("Configure static routes")
+        result = create_static_routes(tgen, static_routes)
+        assert result is True, "Testcase {} : Failed \n Error: {}".format(
+            tc_name, result)
+
+        step("configure redistribute static in Router BGP in R1")
+
+        input_dict_2 = {
+            "r1": {
+                "bgp": {
+                    "address_family": {
+                        addr_type: {
+                            "unicast": {
+                                "redistribute": [{
+                                    "redist_type": "static"
+                                }]
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        result = create_router_bgp(tgen, topo, input_dict_2)
+        assert result is True, "Testcase {} : Failed \n Error: {}".format(
+            tc_name, result)
+
+    step('Configure a route-map on R1 to prepend AS 4 times.')
+    for addr_type in ADDR_TYPES:
+        input_dict_4 = {
+            "r1": {
+                "route_maps": {
+                    "ASP_{}".format(addr_type): [{
+                        "action": "permit",
+                        "set": {
+                            "aspath": {
```

Comment:
please rename `aspath` to `path` because it's already changed in https://github.com/FRRouting/frr/pull/6130 and will be removed

Reply:
and this is valid for `localpref` and `med` if you have those as well somewhere

Reply:
Hi Donatas,

Here we are using "aspath" just to take user input to set attribute for route-map:
input_dict = {
            "r1": {
                "route_maps": {
                    "ASP": [{
                        "action": "permit",
                        "set": {
                            "aspath": {
                                "as_num": "200 200 200 200",
                                "as_action": "prepend"
                            }
                        }
                    }]
                }
            }
        }

# Code from framework, which reads this input data to set as-path prepend:
as_path = set_data.setdefault("aspath", {}).      // Reading aspath dict value

if as_path:
    as_num = as_path.setdefault("as_num", None)              // Reading AS number,  200 200 200 200
    as_action = as_path.setdefault("as_action", None)       // Reading action, prepend from input_dict
    if as_action and as_num:
        rmap_data.append("set as-path {} {}".
                         format(as_action, as_num))         // formed CLI "set as-path prepend 200 200 200 200" as per current input_dict

We will take care of your comment while verifying attributes using "show bgp ipv4/ipv6 json".

Reply:
@ton31337  As you mentioned to bring in following changes:
This change is already since 2018 October.

med --------> metric
localPref --> locPrf
aspath -----> path

I could see that these changes are made to files, but JSON o/p is still showing old attribute names:

show bgp ipv4 ...json o/p:
==================
frr# show bgp ipv4 10.0.20.1/32 json
{
  "prefix":"10.0.20.1\/32",
  "advertisedTo":{
    "10.0.0.14":{
      "hostname":"frr"
    },
    "10.0.0.18":{
      "hostname":"frr"
    }
  },
  "paths":[
    {
      "aspath":{
        "string":"Local",
        "segments":[
        ],
        "length":0
      },
      "origin":"incomplete",
      "med":200,
      "metric":200,
      "localpref":150,
      "valid":true,
      "bestpath":{
        "overall":true,
        "selectionReason":"First path received"
      },
      "lastUpdate":{
        "epoch":1586179727,
        "string":"Mon Apr  6 13:28:47 2020\n"
      },
      "nexthops":[
        {
          "ip":"10.0.0.2",
          "afi":"ipv4",
          "metric":200,
          "accessible":true,
          "used":true
        }
      ],
      "peer":{
        "peerId":"10.0.0.5",
        "routerId":"1.0.1.17",
        "hostname":"frr",
        "type":"internal"
      }
    }
  ]
}

Please check and suggest right format to proceed further.